fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -290,6 +247,16 @@ public CommandProcessingResult createClient(final JsonCommand command) {
 
             final Client newClient = Client.createNew(currentUser, clientOffice, clientParentGroup, staff, savingsProductId, gender,
                     clientType, clientClassification, legalFormValue, command);
+
+            final String externalIdValue = command.stringValueOfParameterNamed(ClientApiConstants.externalIdParamName);
```

Comment:
Use `ExternalIdFactory`. you dont need to to manually these checks!

Reply:
Done